internal/engine/upper.go

```diff
@@ -293,6 +294,7 @@ func handleBuildCompleted(ctx context.Context, engineState *store.EngineState, c
 		}
 
 		ms.LastSuccessfulDeployTime = time.Now()
+		MaybeSetNeedsNudge(mt, engineState)
```

Comment:
it seems very weird that this live in the handleBuildCompletion reducer. why should build completion have to know about analytics?

Reply:
it doesn't have to know, it's just a reasonable place to check for this condition being met -- but you're right that it's more robust just to have a func that checks the engine state, will do that.